FlightTelemetry_2_1_2020/FlightTelemetry_2_1_2020.ino

```diff
@@ -0,0 +1,150 @@
+  SEND_GROUP_ITEM(value.y())          \
+  SEND_GROUP_ITEM(value.z())
+
+#define WRITE_CSV_ITEM(value)         \
+  dataFile.print(F(", ")); dataFile.print(value);
+
+#define WRITE_CSV_VECTOR_ITEM(value)  \
+  WRITE_CSV_ITEM(value.x())           \
+  WRITE_CSV_ITEM(value.y())           \
+  WRITE_CSV_ITEM(value.z())
+
+unsigned int missed_deadlines = 0;
+
+
+void setup() {
+  TELEMETRY_SERIAL.begin(57600);    TELEMETRY_SERIAL.println();
+
+  while (!bmp.begin()) {
+    TELEMETRY_SERIAL.println(F("BMP388 err"));
+    digitalWrite(LED,LOW); delay(1000); digitalWrite(LED,HIGH);
+  } 
+  
+  //does it make sense that the CS pin for the SD card is flashing on/off????
+
+  bmp.setTemperatureOversampling(BMP3_OVERSAMPLING_8X);
+  bmp.setPressureOversampling(BMP3_OVERSAMPLING_4X);
+  bmp.setIIRFilterCoeff(BMP3_IIR_FILTER_COEFF_3);
+
+  //put GPS initialization code here:
+
+
+
+  
+  
+  while (!bno.begin()) {                            // flashes to signal error
+    TELEMETRY_SERIAL.println(F("BNO055 err"));
+    digitalWrite(LED,LOW); delay(1000); digitalWrite(LED,HIGH);
+  }
+  if (!rtc.isrunning()) { rtc.adjust(DateTime(__DATE__, __TIME__)); }
+  if (!SD.begin(BUILTIN_SDCARD)) { TELEMETRY_SERIAL.println(F("SD err")); }
+  else {                                            // generates file name
+    for (uint16_t nameCount = 0; nameCount < 1000; nameCount++) {
+      filename[4] = nameCount/100 + '0';
+      filename[5] = (nameCount%100)/10 + '0';
+      filename[6] = nameCount%10 + '0';
+      if (!SD.exists(filename)) {                   // opens if file doesn't exist
+        dataFile = SD.open(filename, FILE_WRITE);
+        TELEMETRY_SERIAL.print(F("\twriting "));
+        TELEMETRY_SERIAL.println(filename);
+        dataFile.println(F("abs time,sys date,sys time,x_euler_angle,y_euler_angle,z_euler_angle,x_gyro,y_gyro,z_gyro,temperature,x_magnetometer,y_magnetometer,z_magnetometer,x_acceleration,y_acceleration,z_acceleration"));
+        break;
+      }
+    }
+  }
+  
+}
+
+
+void loop() {
+  long time0 = millis();
+  imu::Vector<3> gyroscope     = bno.getVector(Adafruit_BNO055::VECTOR_GYROSCOPE);
+  imu::Vector<3> euler         = bno.getVector(Adafruit_BNO055::VECTOR_EULER);
+  imu::Vector<3> accelerometer = bno.getVector(Adafruit_BNO055::VECTOR_ACCELEROMETER);
+  imu::Vector<3> magnetometer  = bno.getVector(Adafruit_BNO055::VECTOR_MAGNETOMETER);
+  int8_t temp = bno.getTemp();
+  
+  int8_t BMPtemp= bmp.temperature; //in C, do I need int8_t type????
+  //int8_t BMPpressure= bmp.pressure / 100.0; //hPa not super important
+  int8_t BMPalt= bmp.readAltitude(SEALEVELPRESSURE_HPA); //m
+  
+  
+  
+  
+    
+  // Downlink
+  BEGIN_SEND
+  SEND_VECTOR_ITEM(euler_angle ,  euler);
+  SEND_VECTOR_ITEM(gyro        ,  gyroscope);
+  SEND_ITEM(temperature, temp);
+  SEND_VECTOR_ITEM(magnetometer,  magnetometer);
+  SEND_VECTOR_ITEM(acceleration,  accelerometer);
+
+  SEND_ITEM(BMP_Temp, BMPtemp); //can the first part be named whatever I want????

```

Comment:
Yes, but I recommend making them lowercase for consistency.  The name used here just has to match what you add in the telemetry interface spec.

Reply:
gotcha